R/pkg/R/functions.R

```diff
@@ -2452,11 +2453,18 @@ setMethod("date_format", signature(y = "Column", x = "character"),
 #'}
 #' @note from_json since 2.2.0
 setMethod("from_json", signature(x = "Column", schema = "structType"),
-          function(x, schema, ...) {
+          function(x, schema, asArray = FALSE, ...) {
+            if (asArray) {
+              jschema <- callJStatic("org.apache.spark.sql.api.r.SQLUtils",
+                                     "createArrayType",
```

Comment:
It seems we'd better use 

`DataTypes.createArrayType`

per https://github.com/apache/spark/blob/04ee8cf633e17b6bf95225a8dd77bf2e06980eb3/sql/catalyst/src/main/scala/org/apache/spark/sql/types/ArrayType.scala#L49-L56

Let me remove that new wrapper and use the original one.